homeassistant/components/mqtt/notify.py

```diff
@@ -381,15 +377,23 @@ async def async_send_message(self, message: str = "", **kwargs):
         )
 
 
-async def _update_device(
-    hass: HomeAssistant, config_entry: ConfigEntry, config: MqttNotificationConfig
-) -> None:
+def _update_device(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry | None,
+    config: MqttNotificationConfig,
+) -> str | None:
     """Update device registry."""
-    device_registry = await hass.helpers.device_registry.async_get_registry()
+    if config_entry is None or CONF_DEVICE not in config:
+        return None
+
+    device = None
+    device_registry = dr.async_get(hass)
     config_entry_id = config_entry.entry_id
     device_info = device_info_from_config(config[CONF_DEVICE])
 
     if config_entry_id is not None and device_info is not None:
-        update_device_info = dict(device_info)
+        update_device_info = cast(dict, device_info)
```

Comment:
Why do we cast here?

Reply:
I cannot update device info directly to pass that to unpack:
```
homeassistant/components/mqtt/notify.py:395: error: TypedDict "DeviceInfo" has no key "config_entry_id"  [typeddict-item]
homeassistant/components/mqtt/notify.py:396: error: Name "update_device_info" is not defined  [name-defined]
```
So I need to cast it to a dict, to enable to unpack.
Without cast:
```
homeassistant/components/mqtt/notify.py:397: error: Argument 1 to "async_get_or_create" of "DeviceRegistry" has incompatible type "**Dict[str, object]"; expected "str"  [arg-type]
homeassistant/components/mqtt/notify.py:397: error: Argument 1 to "async_get_or_create" of "DeviceRegistry" has incompatible type "**Dict[str, object]"; expected "Union[str, None, UndefinedType]"  [arg-type]
homeassistant/components/mqtt/notify.py:397: error: Argument 1 to "async_get_or_create" of "DeviceRegistry" has incompatible type "**Dict[str, object]"; expected "Optional[Set[Tuple[str, str]]]"  [arg-type]
homeassistant/components/mqtt/notify.py:397: error: Argument 1 to "async_get_or_create" of "DeviceRegistry" has incompatible type "**Dict[str, object]"; expected "Union[DeviceEntryDisabler, None, UndefinedType]"  [arg-type]
homeassistant/components/mqtt/notify.py:397: error: Argument 1 to "async_get_or_create" of "DeviceRegistry" has incompatible type "**Dict[str, object]"; expected "Union[DeviceEntryType, None, UndefinedType]"  [arg-type]
homeassistant/components/mqtt/notify.py:397: error: Argument 1 to "async_get_or_create" of "DeviceRegistry" has incompatible type "**Dict[str, object]"; expected "Optional[Tuple[str, str]]"  [arg-type]
```